Description: detect and use the system's libdouble-conversion
 This patch makes qtdeclarative look for libdouble-conversion. If
 it finds it it will use the system provided version instead of the embedded
 one.
Author: Lisandro Damián Nicanor Pérez Meyer <lisandro@debian.org>
Forwarded: https://codereview.qt-project.org/#/c/105650/

---
 config.tests/double-conversion/double-conversion.pro |    6 ++
 config.tests/double-conversion/main.cpp              |   43 +++++++++++++++++++
 qtdeclarative.pro                                    |    6 ++
 src/qml/jsruntime/jsruntime.pri                      |    7 ++-
 src/qml/jsruntime/qv4numberobject.cpp                |    5 ++
 src/qml/jsruntime/qv4runtime.cpp                     |    6 ++
 6 files changed, 71 insertions(+), 2 deletions(-)

--- /dev/null
+++ b/config.tests/double-conversion/double-conversion.pro
@@ -0,0 +1,6 @@
+CONFIG -= qt
+CONFIG += console
+
+SOURCES += main.cpp
+
+LIBS += -ldouble-conversion
--- /dev/null
+++ b/config.tests/double-conversion/main.cpp
@@ -0,0 +1,43 @@
+/****************************************************************************
+**
+** Copyright (C) 2015 Lisandro Damián Nicanor Pérez Meyer
+**                    <perezmeyer@gmail.com>
+** Contact: http://www.qt-project.org/legal
+**
+** This file is part of the QtQml module of the Qt Toolkit.
+**
+** $QT_BEGIN_LICENSE:LGPL21$
+** Commercial License Usage
+** Licensees holding valid commercial Qt licenses may use this file in
+** accordance with the commercial license agreement provided with the
+** Software or, alternatively, in accordance with the terms contained in
+** a written agreement between you and Digia. For licensing terms and
+** conditions see http://qt.digia.com/licensing. For further information
+** use the contact form at http://qt.digia.com/contact-us.
+**
+** GNU Lesser General Public License Usage
+** Alternatively, this file may be used under the terms of the GNU Lesser
+** General Public License version 2.1 or version 3 as published by the Free
+** Software Foundation and appearing in the file LICENSE.LGPLv21 and
+** LICENSE.LGPLv3 included in the packaging of this file. Please review the
+** following information to ensure the GNU Lesser General Public License
+** requirements will be met: https://www.gnu.org/licenses/lgpl.html and
+** http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html.
+**
+** In addition, as a special exception, Digia gives you certain additional
+** rights. These rights are described in the Digia Qt LGPL Exception
+** version 1.1, included in the file LGPL_EXCEPTION.txt in this package.
+**
+** $QT_END_LICENSE$
+**
+****************************************************************************/
+
+#include <double-conversion/double-conversion.h>
+
+using namespace double_conversion;
+
+int main(int, char**)
+{
+    DoubleToStringConverter(0,"inf","nan",'e',-2,5,3,3);
+    return 0;
+}
--- a/qtdeclarative.pro
+++ b/qtdeclarative.pro
@@ -1,3 +1,9 @@
 CONFIG += tests_need_tools examples_need_tools
+
+linux {
+    load(configure)
+    qtCompileTest(double-conversion)
+}
+
 load(qt_parts)
 
--- a/src/qml/jsruntime/jsruntime.pri
+++ b/src/qml/jsruntime/jsruntime.pri
@@ -113,4 +113,9 @@
 
 ios: DEFINES += ENABLE_ASSEMBLER_WX_EXCLUSIVE=1
 
-include(../../3rdparty/double-conversion/double-conversion.pri)
+config_double-conversion {
+    DEFINES += SYSTEM_DOUBLECONVERSION
+    LIBS_PRIVATE += -ldouble-conversion
+} else {
+    include(../../3rdparty/double-conversion/double-conversion.pri)
+}
--- a/src/qml/jsruntime/qv4numberobject.cpp
+++ b/src/qml/jsruntime/qv4numberobject.cpp
@@ -39,7 +39,12 @@
 #include <QtCore/qmath.h>
 #include <QtCore/QDebug>
 #include <cassert>
+
+#ifdef SYSTEM_DOUBLECONVERSION
+#include <double-conversion/double-conversion.h>
+#else
 #include <double-conversion.h>
+#endif
 
 using namespace QV4;
 
--- a/src/qml/jsruntime/qv4runtime.cpp
+++ b/src/qml/jsruntime/qv4runtime.cpp
@@ -60,7 +60,11 @@
 
 #include <wtf/MathExtras.h>
 
-#include "../../3rdparty/double-conversion/double-conversion.h"
+#ifdef SYSTEM_DOUBLECONVERSION
+#include <double-conversion/double-conversion.h>
+#else
+#include <double-conversion.h>
+#endif
 
 #ifdef QV4_COUNT_RUNTIME_FUNCTIONS
 #  include <QtCore/QBuffer>
